src/Umbraco.Infrastructure/Manifest/ManifestParser.cs

```diff
@@ -250,6 +250,11 @@ private IEnumerable<string> GetManifestFiles()
             return Array.Empty<string>();
         }
 
-        return Directory.GetFiles(_path, "package.manifest", SearchOption.AllDirectories);
+        var files = Directory.GetFiles(_path, "package.manifest", SearchOption.AllDirectories);
+
+        // Ensure a consistent sorting of paths
+        Array.Sort(files);
```

Comment:
Might be good to include a more verbose explanation why this is needed, e.g.:

```suggestion
        // Ensure a consistent, alphabetical sorting of paths, because this is not guaranteed to be the same between file systems or OSes
        Array.Sort(files);
```

Also note that in most cases (sub)folders are returned before the files, but since this uses `SearchOption.AllDirectories` and sorts on the whole path, this might result in a different (maybe unexpected) ordering, e.g.:
1. `App_Plugins\MyPackage\BackOffice\MyEditor\package.manifest`
2. `App_Plugins\MyPackage\package.manifest`
3. `App_Plugins\MyPackage\Search\package.manifest`

And just for reference, the [Microsoft documentation](https://learn.microsoft.com/en-us/dotnet/api/system.io.directory.getfiles?view=net-7.0#system-io-directory-getfiles(system-string)) does mention this:
> The order of the returned file names is not guaranteed; use the [Sort](https://learn.microsoft.com/en-us/dotnet/api/system.array.sort?view=net-7.0) method if a specific sort order is required.